src/mixtures/mixturemodel.jl

```diff
@@ -500,6 +500,21 @@ componentwise_logpdf(d::MultivariateMixture, x::AbstractVector) = componentwise_
 componentwise_logpdf(d::MultivariateMixture, x::AbstractMatrix) = componentwise_logpdf!(Matrix{eltype(x)}(undef, size(x,2), ncomponents(d)), d, x)
 
 
+function quantile(d::UnivariateMixture{Continuous}, p::Real)
+    ps = probs(d)
+    cs = [component(d,i) for i in eachindex(ps) if ps[i] > 0.0]
+    qs = quantile.(cs, p)
+    min_q, max_q = extrema(qs)
```

Comment:
An alternative without allocating `cs` and `qs`:
```suggestion
    min_q, max_q = extrema(quantile(component(d, i), p) for i in 1:ncomponents(d) if ps[i] > 0)
```